src/Nancy/AfterPipeline.cs

```diff
@@ -83,6 +83,12 @@ private static void ExecuteTasksInternal(NancyContext context, CancellationToken
             {
                 var current = enumerator.Current.Invoke(context, cancellationToken);
 
+                if (current == null)
+                {
+                    var method = enumerator.Current.Method ?? (object)enumerator.Current;
+                    throw new InvalidOperationException(string.Format("'{0}' returned null; a non-null {1} expected.", method, typeof(Task)));

```

Comment:
Do you have an example of this error message? Both for the `Task.Method` and the `Task` itself. Something's telling me that this won't be very helpful :confused: